common/scripts/x86_64/CentOS_7/_helpers.sh

```diff
@@ -339,6 +343,57 @@ __pull_images() {
   done
 }
 
+__pull_admiral_image() {
+  __process_marker "Pulling latest admiral image"
+  __process_msg "Registry: $PRIVATE_IMAGE_REGISTRY"
+  __registry_login
+
+  image="$PRIVATE_IMAGE_REGISTRY/$ADMIRAL_IMAGE:$RELEASE"
+  sudo docker pull $image
+}
+
+__pull_images_master() {
+  __process_marker "Pulling latest service images on master"
+  __process_msg "Registry: $PRIVATE_IMAGE_REGISTRY"
+  __registry_login
+
+  local master_ip=""
+
+  if [ "$DEV_MODE" != "true" ]; then
+    local get_master_query="PGPASSWORD=$DB_PASSWORD \
+      psql \
+      -U $DB_USER \
+      -d $DB_NAME \
+      -h $DB_IP \
+      -p $DB_PORT \
+      -v ON_ERROR_STOP=1 \
+      -tc 'SELECT master from \"systemSettings\"; '"
+
+    {
+      local master=`eval $get_master_query` &&
+      __process_msg "'systemSettings' table exists, finding master IP"
+      master_ip=$(echo "$master" | jq '.address')
+    } || {
+      __process_msg "'systemSettings' table does not exist, skipping"
+    }
+  fi
+
+  for image in "${SERVICE_IMAGES[@]}"; do
+    image="$PRIVATE_IMAGE_REGISTRY/$image:$RELEASE"
+    local pull_cmd="sudo docker pull $image"
+    if [ -z "$master_ip" ]; then
+      __process_msg "Pulling $image"
+      sudo docker pull $image
+    else
+      local docker_login_cmd="aws ecr --no-include-email --region us-east-1 get-login | bash"
```

Comment:
why are we not using the docker login helper function to do this. again we'll end up in a situation where this logic is scattered all across the codebase and cannot be changed.

Reply:
my bad. this executes on remote thats why i suppose. But again, the command that gets constructed here should be centralized in one function since we use it in multiple places

Reply:
this taken from the previous flow of how we were doing it. I would like to clean it, but because of time contraint I implemented whatever was there before. Probably, I could do it after this release?

Reply:
ok. lets make a note of it then here under a `TODO`. we should try to clean things up with subsequent iterations on this file.

Reply:
added